akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala

```diff
@@ -81,56 +109,46 @@ import org.slf4j.LoggerFactory
   override def getSystem: akka.actor.typed.ActorSystem[Void] =
     system.asInstanceOf[ActorSystem[Void]]
 
-  override def log: Logger = {
-    val l = logger match {
+  private def loggingContext(): LoggingContext = {
+    // lazy init of logging setup
+    _logging match {
       case OptionVal.Some(l) => l
       case OptionVal.None =>
         val logClass = LoggerClass.detectLoggerClassFromStack(classOf[Behavior[_]])
-        initLoggerWithName(logClass.getName)
+        val logger = LoggerFactory.getLogger(logClass.getName)
+        val l = new LoggingContext(logger, classicActorContext.props.deploy.tags)
+        _logging = OptionVal.Some(l)
+        l
     }
-    if (tags.isEmpty) initTags()
-    // avoid access to MDC ThreadLocal if not needed
-    // FIXME mdc is always used if logger is initialized, so we could use that as a flag instead and save one field
-    mdcUsed = true
-    ActorMdc.setMdc(self.path.toString, tags)
-    l
+  }
+
+  override def log: Logger = {
+    val logging = loggingContext()
+    logging.mdcUsed = true
```

Comment:
the "// avoid access to MDC ThreadLocal if not needed" comment was lost. Maybe add some more based on my explanation.

Reply:
Moved explanation into the context class, but I'll re-add it here as well.